src/main/java/frc/robot/components/MachineLearning.java

```diff
@@ -0,0 +1,181 @@
+    @param cmd the command you want to run
+    @param startingDir the starting directory of the command
+    @return the output of the command
+    */
+    private String executeCmd(String cmd, String startingDir) {
+        if(startingDir == ""){
+            startingDir = "/";
+        }
+
+        String[] split = cmd.split(" ");
+        String[] command = new String[split.length + 1];
+        String out = "";
+
+        if(isUnix()) {
+            command[0] = "sudo";
+        }
+
+        else {
+            command[0] = "powershell";
+        }
+
+        for(int i = 1; i < command.length; i++) {
+            command[i] = split[i - 1].replaceAll("//s"," ");
+        }
+
+        ProcessBuilder builder = new ProcessBuilder(command);
+        builder = builder.directory(new File(startingDir));
+
+        try {
+            Process p = builder.start();
+            BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(p.getInputStream()), 1);
+            String line = null;
+
+            while ((line = bufferedReader.readLine()) != null) {
+                out += line + "\n";
+            }
+        }
+
+        catch(IOException e) {
+            e.printStackTrace();
+        }
+
+        return out;
+    }
+
+    public MachineLearning(boolean record, String filename) {
+        this.filename = filename + ".txt";
+        forwardSpeed.clear();
+        sideSpeed.clear();
+        frontLeftEncoderPosition.clear();
+        frontRightEncoderPosition.clear();
+        backLeftEncoderPosition.clear();
+        backRightEncoderPosition.clear();
+
+        if(record) {
+            executeCmd("mkdir /paths");
+            executeCmd("chmod a=rwx /paths");
+
+            try {
+                File myObj = new File("/paths/" + filename);
+
+                if (myObj.createNewFile())  {
+                    System.out.println("File created: " + myObj.getName());
+                } 
+                
+                else  {
+                    System.out.println("File already exists.");
+                }
+            } 
+            catch (IOException e) {
+                System.out.println("An error occurred.");
+                e.printStackTrace();
+            }
+        }
+    }
+
+    public void init() {
+        forwardSpeed.clear();
+        sideSpeed.clear();
+        frontLeftEncoderPosition.clear();
+        frontRightEncoderPosition.clear();
+        backLeftEncoderPosition.clear();
+        backRightEncoderPosition.clear();
+    }
+
+    public void periodic(Double _forwardSpeed, Double _sideSpeed, Double flep, Double frep, Double blep, Double brep) {
+        if(forwardSpeed.size() <= MAX_SIZE) {
+            forwardSpeed.add(_forwardSpeed);
+            sideSpeed.add(_sideSpeed);
+            frontLeftEncoderPosition.add(flep);
+            frontRightEncoderPosition.add(frep);
+            backLeftEncoderPosition.add(blep);
+            backRightEncoderPosition.add(brep);
```

Comment:
If we're writing to a file, should we write the line immediately? What's the value in buffering to the ArrayLists?